lib/flat_search/olx_scraper.ex

```diff
@@ -1,28 +1,46 @@
-defmodule OlxScraper do
-  use Crawly.Spider
+defmodule FlatSearch.OlxScraper do
   alias FlatSearch.Flats
 
-  @impl Crawly.Spider
-  def base_url(), do: "https://www.olx.pl"
+  def run do
+    url = "https://www.olx.pl/nieruchomosci/mieszkania/wynajem/"
+    res = Crawly.fetch(url)
+    {:ok, document} = Floki.parse_document(res.body)
 
-  @impl Crawly.Spider
-  def init(), do: [start_urls: ["https://www.olx.pl/nieruchomosci/mieszkania/wynajem/"]]
+    # Run get_flat() for every page simultaneously
+    for n <- get_range_of_pages(document) do
+      Task.async(fn -> get_flats(url <> "?page=#{n}") end)
+    end
+  end
 
-  @impl Crawly.Spider
-  def parse_item(res) do
-    # First page only
+  defp get_flats(url) do
+    res = Crawly.fetch(url)
     {:ok, document} = Floki.parse_document(res.body)
 
     flat_urls =
       document
       |> Floki.find("tr td div table tbody tr td a.link")
       |> Floki.attribute("href")
-      |> Enum.filter(&String.contains?(&1, base_url()))
+      |> Enum.filter(&String.contains?(&1, "https://www.olx.pl"))
       |> Enum.uniq()
 
     Enum.map(flat_urls, &parse_flat(&1))
   end
 
+  defp get_range_of_pages(document) do
+    # Gets number of pages
+    num_of_pages =
+      document
+      |> Floki.find("[data-cy=page-link-last]")
+      |> Enum.at(0)
+      |> elem(2)
+      |> Enum.at(0)
+      |> elem(2)
+      |> Enum.at(0)
+      |> str_to_num()
+
+    1..num_of_pages
```

Comment:
Maybe change this range creating `1..num_of_pages` to the function at the end of the pipeline. ie `|> Range.new(1)`, it would iterate from the oldest to the newest page tho